arangod/IResearch/IResearchAqlAnalyzer.cpp

```diff
@@ -238,6 +239,12 @@ class CalculationTransactionState final : public arangodb::TransactionState {
     return {};
   }
 
+  [[nodiscard]] arangodb::Result performIntermediateCommitIfRequired(
```

Comment:
Does it have to be nodiscard if you're getting nothing from the result?

Reply:
`nodiscard` means that the function return value must be somehow handled by the caller.
It doesn't matter for the compiler what's in the return value. Just that it is not immediately discarded at the call site.

Reply:
yeah but then you'd have to do something with the returned value like store in a variable so it doesn't complain